drivers/timer/xtensa_sys_timer.c

```diff
@@ -111,6 +119,19 @@ uint32_t sys_clock_cycle_get_32(void)
 	return ccount();
 }
 
+uint64_t sys_clock_cycle_get_64(void)
+{
+	uint32_t diff;
+	uint32_t curr = ccount();
+
+	if (curr < last_count)
+		diff = (MAX_CYC - last_count) + curr;
+	else
+		diff = curr - last_count;
```

Comment:
This needs locking, otherwise you can take an interrupt here and update last_count after having read your counter.  Also same issue with the manual wrapping math.